core/src/main/java/org/gbif/pipelines/core/functions/interpretation/InterpretationException.java

```diff
@@ -0,0 +1,46 @@
+package org.gbif.pipelines.core.functions.interpretation;
+
+import org.gbif.pipelines.core.functions.interpretation.error.Issue;
+import org.gbif.pipelines.core.functions.interpretation.error.Lineage;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Interpretation Exception with issues and lineages
+ * also set interpretedObject in case you had one
+ */
+public class InterpretationException extends Exception {
+
+  private List<Issue> issues = new ArrayList<>();
+  private List<Lineage> lineages = new ArrayList<>();
+  private Object interpretedValue;
```

Comment:
can Object be replaced by a type parameter InterpretationException<T> and maybe use 
private Optional<T> interpretedValue, this is only a suggestion and we should evaluate its implications